File: src/dataset/cpp/prime_factors_Exercism_johnngugi/gemini/GeminiFlash_prime_factors_v4.cpp
#include "prime_factors.h"
#include <vector>

namespace prime_factors
{
    std::vector<int> of(int n)
    {
        std::vector<int> factors;

        if (n <= 1) {
            return factors;
        }

        while (n % 2 == 0) {
            factors.push_back(2);
            n /= 2;
        }

        for (int i = 3; i * i <= n; i += 2) {
            while (n % i == 0) {
                factors.push_back(i);
                n /= i;
            }
        }

        if (n > 1) {
            factors.push_back(n);
        }

        return factors;
    }
}

File: src/dataset/javascript/prime-factors_exercism-javascript-ThomasZumsteg/gemini/GeminiFlash_prime-factors_v4.js
function primeFactors(remainder) {
  /* Finds all prime factors of a number */
  // Finds factors smallest to largest adding to the list of factors
  const factors = [];

  if (remainder <= 1) {
    return factors;
  }

  while (remainder % 2 === 0) {
    factors.push(2);
    remainder /= 2;
  }

  let factor = 3;
  const sqrtRemainder = Math.sqrt(remainder);

  while (factor <= sqrtRemainder && remainder > 1) {
    while (remainder % factor === 0) {
      factors.push(factor);
      remainder /= factor;
    }
    factor += 2;
  }

  if (remainder > 1) {
    factors.push(remainder);
  }

  return factors;
}

export default { for: primeFactors };